omop_file_validator.py

```diff
@@ -176,6 +180,32 @@ def find_blank_lines(f):
     return [i + 1 for i in indices]
 
 
+def find_scientific_notation_errors(f, int_columns):
+    df = pd.read_csv(f, dtype=str)
+    df = df.rename(columns=str.lower)
+    df = df[[col for col in int_columns if col in df.columns]]
+
+    errors = []
+    sci_not_line = collections.defaultdict(int)
+
+    for submission_col_name in df.columns:
+        submission_column = df[submission_col_name]
+        for i, value in submission_column.items():
+            if pd.notnull(value) and re.match(SCIENTIFIC_NOTATION_REGEX,
+                                              value):
+                sci_not_line[submission_col_name] = i + 1
+                break
+
+    for col, linenum in sci_not_line.items():
+        e = dict(message=f"Scientific notation was found on line {linenum}. " \
+                            "Scientific notation is not allowed for integer fields.",
+                column_name=col
```

Comment:
One small improvement would be to catch the erroneous `value` as well, maybe in a tuple
```suggestion
                sci_not_line[submission_col_name] = (value, i + 1)
                break

    for col, (value, line_num) in sci_not_line.items():
        e = dict(message=(f"Scientific notation '{value}' was found on line {line_num}. "
                      "Scientific notation is not allowed for integer fields."),
                column_name=col
```

Reply:
Made another addition